### PR TITLE
[WIP] Fix more migration bugs [#OSF-7422]

### PR DESCRIPTION
**Pull request by cwisecarver**
## Purpose

This fixes a bunch of bugs found when running migrations against production data. It also adds validation mgmt commands for confirming the migration ran correctly.

## Changes

- Add order field to node_relations
- Force node titles to be <= 200char
- Switch from print to logger.info
- Fix dropbox repr
- Stop indexing the location field, production data is too big
- Add code to try and recover guids with None referents, guids with inaccurate referents, etc

## Side effects

None are obvious.

### CODE REVIEW

**Comment by cwisecarver** [comment]
@sloria I'm pretty sure this is safe but now that we're in develop, if you wanted to look it over before it's merged that would be appreciated.

File: osf_tests/settings.py [hunk 1]
```diff
@@ -22,7 +22,6 @@
 
 REST_FRAMEWORK['ALLOWED_VERSIONS'] = (
         '2.0',
-        '2.0.1',
```
Comment (sloria) [inline osf_tests/settings.py]: Why was this removed?

File: osf/management/commands/migratedata.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import importlib
 import itertools
 import logging
+import pprint
```
Comment (sloria) [inline osf/management/commands/migratedata.py]: This doesn't appear to be used.

File: osf/management/commands/migraterelations.py [hunk 3]
```diff
@@ -437,79 +445,87 @@ def migrate_node_through_models(self):
...
+                            if count % page_size == 0 or count == total:
+                                Contributor.objects.bulk_create(contributors)
+                                logger.info('Through {} nodes and {} contributors, '
+                                      'saved {} contributors'.format(count, contributor_count, len(contributors)))
+                                contributors = []
+                                modm_obj._cache.clear()
+                                modm_obj._object_cache.clear()
+                                logger.info('Took out {} trashes'.format(gc.collect()))
+
+                            node_relations = []
```
Comment (felliott) [inline osf/management/commands/migraterelations.py]: I think this should be declared in the outer scope with `contributors`.  Otherwise, only `node_relations` for the last page of nodes (if there are any) will be created.

File: osf/management/commands/migratedata.py (hunk 2 above)
Comment (sloria) [inline osf/management/commands/migratedata.py]: Oh woops, this is outdated... ignore

File: osf/management/commands/migraterelations.py (hunk 3 above)
Comment (cwisecarver) [inline osf/management/commands/migraterelations.py]: You're right. I'll fix it. Seems like a very strange loop structure. I know it saves iterations but doing two individual things inside one loop iteration is confusing. I might pull this out into methods or something just to make it easier to not be very confused by.

File: osf_tests/settings.py (hunk 1 above)
Comment (cwisecarver) [inline osf_tests/settings.py]: Not sure. Either a bad merge or a quick trigger finger. It's still in local_travis so I'll put it back here.

**Comment by cwisecarver** [comment]
closing in favor of #6818
